### PR TITLE
chore: Add HPO support to readme and SDK diagram

### PR DESCRIPTION
**Pull request by kramaranya**
I've updated README with HPO support and added a new Kubeflow SDK diagram

/assign @kubeflow/kubeflow-sdk-team

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +21,10 @@ ML applications rather than managing complex infrastrutcure.
 
 <div style="text-align: center;">
   <img
-    src="https://raw.githubusercontent.com/kubeflow/sdk/main/docs/images/persona_diagram.svg"
+    src="https://raw.githubusercontent.com/kubeflow/sdk/main/docs/images/kubeflow-sdk.drawio.svg"
```
Comment (andreyvelich) [inline README.md:24]: You can remove this file: https://github.com/kubeflow/sdk/blob/main/docs/images/persona_diagram.svg

File: README.md [hunk 2]
```diff
@@ -39,9 +39,9 @@ pip install -U kubeflow
 ### Run your first PyTorch distributed job
 
 ```python
-from kubeflow.trainer import TrainerClient, CustomTrainer
+from kubeflow.trainer import TrainerClient, CustomTrainer, TrainJobTemplate
 
-def get_torch_dist():
+def get_torch_dist(learning_rate, num_epochs):
```
Comment (andreyvelich) [inline README.md]:
```suggestion
def get_torch_dist(learning_rate: str, num_epochs: str):
```

File: README.md [hunk 3]
```diff
@@ -52,33 +52,61 @@ def get_torch_dist():
...
     print(f"LOCAL_RANK: {os.environ['LOCAL_RANK']}")
 
-# Create the TrainJob
-job_id = TrainerClient().train(
+    lr = float(learning_rate)
+    epochs = int(num_epochs)
+    accuracy = 0.8 + (lr * 10) + (epochs * 0.01)
+
+    if dist.get_rank() == 0:
+        print(f"accuracy={accuracy}")
```
Comment (andreyvelich) [inline README.md]: Maybe you can use `loss` in this example, to just rely on default object and reduce number of parameters you need to set in `optimize` API ?
https://github.com/kubeflow/sdk/blob/b5626160d7e376e1ba662501df569b35fadb61e1/kubeflow/optimizer/types/optimization_types.py#L44

**Review by andreyvelich — COMMENTED** [review]
Thanks for this @kramaranya!

File: README.md (hunk 3 above)
Comment (kramaranya) [reply README.md]: Sure, updated in https://github.com/kubeflow/sdk/pull/141/commits/857590f5a300aed655f894e4e2370f5f271f712a

**Comment by astefanutti** [comment]
/lgtm

Thanks!
